blockmanager.go

```diff
@@ -452,9 +447,8 @@ func (b *blockManager) handleTxMsg(tmsg *txMsg) {
 
 	// Process the transaction to include validation, insertion in the
 	// memory pool, orphan handling, etc.
-	allowOrphans := cfg.MaxOrphanTxs > 0
 	acceptedTxs, err := b.txMemPool.ProcessTransaction(tmsg.tx,
-		allowOrphans, true, mempool.Tag(tmsg.peer.ID()))
```

Comment:
This changes the behavior slightly. Previously, if `MaxOrphanTxs` was configured to be 0 and the node received an orphan transaction, it would respond with a reject (duplicate) packet. Now it will just not add the tx to the orphan pool and silently carry on. This seems more like the right behavior.

There are ways to modify the logic to achieve parity with Core (eg. request parent txs), but that can be added later.